cpu: x64: ukernel: transform: add incorrect ldb check (#2436)
oneapi-src/oneDNN: doc/ukernel/operations/transform.md

=======================================

>
> [API Reference](@ref dnnl::ukernel::transform)
>

## General

The [BRGeMM ukernel](@ref dev_guide_ukernel_brgemm) might require the B tensor
in a specific memory layout depending on target data types and the machine
architecture. Users should check the requirement by calling the

> [get_B_pack_type()](@ref dnnl::ukernel::brgemm::get_B_pack_type) function. If it
> returned the [pack32](@ref dnnl::ukernel::pack_type::pack32) type, the packing

Reviewer: ```suggestion
[get_B_pack_type()](@ref dnnl::ukernel::brgemm::get_B_pack_type) function. If it
returns the [pack32](@ref dnnl::ukernel::pack_type::pack32) type, the packing
```

is required, otherwise, it's not.

The transform ukernel allows users to convert data from the original layout

> which is described as either

Author: Thanks for the edits, incorporated.

> which are described as either

Reviewer: ```suggestion
which is described as either
```

## Implementation limitations

> - Destination leading dimension, or `out_ld`, only supported values are: `16`,

Reviewer: ```suggestion
- Destination leading dimension, or `out_ld`, must be one of the following values: `16`,
```

>   `32`, `48`, or `64`. This is the implementation limitation, there's no

Reviewer: ```suggestion
  `32`, `48`, or `64`. This is the implementation limitation, there are no
```

>   effecient kernels supported for other leading dimension values.

Reviewer: ```suggestion
  efficient kernels supported for other leading dimension values.
```

Reviewer: Could you please clarify if there are no efficient kernels or kernels at all for other leading dimensions?

Author: Thanks for the suggestions, incorporated.

From the end-user perspective it should be no difference whether there's no kernels or they are not efficient at all. In theory we can plug simple_reorder in, but it would be no point in doing that. It's just a comment explaining where the limitation is coming from.
